apps/angular/src/app/features/anime/table/table.component.ts

```diff
@@ -23,11 +71,101 @@ export class TableComponent {
 		'status',
 	];
 
-	/** List of anime. */
-	protected readonly animeList$ = this.getAnimeList();
+	public constructor(private fb: FormBuilder) {
+		this.makeSnapshots();
+
+		this.animeList$ = combineLatest([
+			this.pagination$,
+			this.sorting$,
+			this.filterForm$,
+		]).pipe(
+			map(([pagination, sorting, filter]) => ({ pagination, sorting, filter })),
+			tap(params => this.setQueryParams(params)),
+			switchMap(param => this.animeService.getAnimeList(param)),
+		);
+	}
+
+	/**
+		* Makes snapshot of query params.
+		* Saves the value of the parameters if the parameters exist.
+	 */
+	private makeSnapshots(): void {
+		const snapshot = this.route.snapshot.queryParams;
+
+		if (snapshot['pageIndex'] && snapshot['pageSize']) {
```

Comment:
```suggestion
		if (snapshot['pageIndex'] != null && snapshot['pageSize']  != null ) {
```

Avoid type casts.
What if you have a filter with a boolean value? Will this condition work the same way?